cocos/base/ccTypes.h

```diff
@@ -165,6 +165,24 @@ struct CC_DLL Color4F
     static const Color4F GRAY;
 };
 
+Color4F& operator+=(Color4F& lhs, const Color4F& rhs);
+Color4F operator+(Color4F lhs, const Color4F& rhs);
+Color4F& operator-=(Color4F& lhs, const Color4F& rhs);
+Color4F operator-(Color4F lhs, const Color4F& rhs);
+
+Color4F& operator*=(Color4F& lhs, const Color4F& rhs);
+Color4F& operator*=(Color4F& lhs, float rhs);
+template<typename T>
+Color4F operator*(Color4F lhs, const T& rhs) {
+    return lhs *= rhs;
+}
```

Comment:
Why need this function? Used to convert other types to `Color4F` first then invoke `Color4F& operator*=(Color4F& lhs, const Color4F& rhs)`? If so, need to add similar template function for `-` and `+`?